### PR TITLE
Content for Frontend Setup in Base Camp

### PR DESCRIPTION
**Pull request by briandoyle81CB**
**What changed? Why?**
New content

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import { publicProvider } from 'wagmi/providers/public';
+```
+
+:::caution
+If you're adapting this guide to a different set of libraries or platforms, you may need to import `styles.css` differently. You'll know this is the case if you get ugly text at the bottom of the page instead of a nice modal when you click the connect button.
+:::
+
+### Connectors
+
+Next, configure the chains, wallet connectors, and providers for your app. We'll use the `publicProvider` for now, to get started. See our guide on [Connecting to the Blockchain] for more information on blockchain providers.
```
Comment (Zizzamia) [inline apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:100]: Btw, when you use the word `Next`, I wasn't sure if you were refering to Next.js or just saying the word `next`. lol

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+```typescript
+// This function has been simplified to save space.
+export default function Home() {
+  return (
+    <main
+      className={`flex min-h-screen flex-col items-center justify-between p-24 ${inter.className}`}
+    >
+      <div className="z-10 w-full max-w-5xl items-center justify-between font-mono text-sm lg:flex">
+        <ConnectButton />
```
Comment (Zizzamia) [inline apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:174]: This is the simple approach, should we leave a note also on where to learn for customizing the Button? https://www.rainbowkit.com/docs/custom-connect-button

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+[Rainbowkit] is one of several options that makes this a little bit easier by serving as an aggregator of wallets, and handling some of the details of connecting them. Alternatives include [ConnectKit], and [Dynamic], which are both excellent choices as well.
+
+Each of these include customizable UI/UX components for inviting the user to connect, displaying connection status, and selecting which wallet they wish to use.
+
+### Using the Quick Start
+
+If you're just trying to get up and running as quickly as possible, you can use RainbowKit's [quick start] script to scaffold an app from their template, with a single command. If you're using Yarn:
+
+```bash
+yarn create @rainbow-me/rainbowkit
```
Comment (Zizzamia) [inline apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:34]: Why are we recommending this? I mean, what's the advantage to have this vs https://github.com/coinbase/build-onchain-apps, which uses RainbowKit already.

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+```bash
+yarn run dev
+```
+
+Using the script is fast, but it does mean less choice. In this case, it builds the app on top of [Next.js], which is great if you want to use it, but not helpful if you prefer to work from a different framework, such as [Create React App], or [Remix]. The script also doesn't help you if you want to add an onchain integration to an existing site.
+
+---
+
+## Conclusion
```
Comment (Zizzamia) [inline apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:53]: General take, why are we doing Conclusions if we already learn on what we are learning from the Objectives?

I mostly ask becasue when I browse around for docs, I want something easy to read, and streight to the point. Anything too wordy it usually makes me bounce back very fast.

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md (hunk 1 above)
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:100]: Haha, i'll pick a different one.

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+You can now wrap your app with the context providers for RainbowKit and wagmi. This will make your connection to the blockchain available throughout your entire app without needing to pass anything through props.
+
+```typescript
+export default function RootLayout({ children }: { children: React.ReactNode }) {
+  return (
+    <html lang="en">
+      <WagmiConfig config={wagmiConfig}>
+        <RainbowKitProvider chains={chains}>
+          <body className={inter.className}>{children}</body>
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:146]: You know better than me, I'll move them inside!

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+import { configureChains, createConfig, WagmiConfig } from 'wagmi';
+import { mainnet, base, baseGoerli } from 'wagmi/chains';
+import { publicProvider } from 'wagmi/providers/public';
+```
+
+:::caution
+If you're adapting this guide to a different set of libraries or platforms, you may need to import `styles.css` differently. You'll know this is the case if you get ugly text at the bottom of the page instead of a nice modal when you click the connect button.
+:::
+
+### Connectors
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:98]: I always like to link directly to them when relevant. Will add

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md (hunk 3 above)
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:34]: We're not recommending it, we're just using it.  For learning purposes, the RainbowKit template is good because it's much lighter and simpler than our onchain-apps template.  But I will be linking to that in the beginning as a much more robust starter!

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md (hunk 4 above)
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:53]: The conclusion bookends the learning experience and improves longterm retention by having the reader review and remember the content one last time.

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md (hunk 3 above)
Comment (Zizzamia) [reply apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:34]: What's the goal for us, show case a Wallet Connector or kickstart a project? Beacuse the code RainbowKit produce which is this one https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-create-react-app/src is very very very light.

At that point, would just the doc be enough? or access any of their CodeSandbox to just see how it plays out?

**Review by amhayslipcb — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for amhayslipcb @ 2023-11-28 18:08:49 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md (hunk 4 above)
Comment (Zizzamia) [reply apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:53]: I feel like Call To Action at the end are more important than a recap. I personally find recap for a doc that is less than 1k words, not need it.

Recap for books where chapters are like 30 pages, there is need it. As you might have forgot where you start reading.

I understand in theory recap are useful, but I think we disagree in when they should be used. And it's ok we disagree.

File: apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+  );
+}
+```
+
+Run your app with `yarn dev`, and you should be able to use the RainbowKit connect button to connect with your wallet and switch between networks.
+
+---
+
+## Conclusion
+
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md:196]: Yup! 😳

File: apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md (hunk 3 above)
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/frontend-setup/wallet-connectors.md:34]: Grabbed some time to dig into it!

**Review by taycaldwell — APPROVED** [review]

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+
+    for(uint i = 0; i < issues.length; i++) {
+        allIssues[i] = getIssue(i);
+    }
+
+    return allIssues;
+}
+```
+
+**You also need to make the `getIssue` function `public`. The original spec called for it to be `external`.**
```
Comment (CryptoUGuy) [inline apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md:49]: this may not matter a lot since we have a mix of things in the whole documentation but I think that in the most purist way, if we are not calling a function within the contract, should be external no?

**Comment by cb-heimdall** [comment]
Review Error for CryptoUGuy @ 2023-12-11 21:52:17 UTC
User must have write permissions to review

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/useAccount.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+- Implement the `useAccount`` hook to show the user's address, connection state, network, and balance
+- Implement an `isMounted` hook to prevent hydration errors
+
+---
+
+## Displaying Connection Information
+
+We'll be working from an app generated by RainbowKit's [quick start]. Either open the one you created when we were exploring [Wallet Connectors], or create a new one for this project.
+
+Either way, change the list of chains to only include `baseGoerli` as the network option. You don't want to accidentally spend real money while developing!
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/reading-and-displaying-data/useAccount.md:26]: New guidance has been to only include base

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/configuring-useContractRead.md [hunk 10]
```diff
@@ -0,0 +1,276 @@
+---
+title: Configuring `useContractRead`
+description: Configure the properties of the `useContractRead`` hook.
+hide_table_of_contents: false
+---
+
+The [`useContractRead`] hook has a number of configurable properties that will allow you to adapt it to your needs. You can
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/reading-and-displaying-data/configuring-useContractRead.md]: Whoops, thanks!

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md (hunk 8 above)
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md:49]: We're calling it internally with the new `getAllIssues` function I added.  So you can get one by id, or all.

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md [hunk 11]
```diff
@@ -0,0 +1,450 @@
...
+
+:::
+
+### Create Demo Issues
+
+To start, you'll need to put some data into your contract so that you can read it from your frontend. Open [Goerli BaseScan], find your contract, connect with your wallet, and call the `claim` function.
+
+Add the following two issues:
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md:57]: We're just using it temporarily to send transactions so we have something to read.  Seemed like this was the simplest way.

File: apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md [hunk 12]
```diff
@@ -0,0 +1,450 @@
...
+
+For this guide, we'll be continuing from the project you started for the [`useAccount` hook]. We'll work with an upgrade to the contract that you may have created if you completed the [ERC 20 Tokens Exercise]. See below for an example you can use if you don't already have your own!
+
+The contract creates a very simple DAO, in which users can create issues and vote for them, against them, or abstain. Anyone can `claim` 100 tokens. This is an insecure system for demonstration purposes, since it would be trivial to claim a large number of tokens with multiple wallets, then transfer them to a single address and use that to dominate voting.
+
+But it makes it much easier to test!
+
+:::caution
+
+If you're using your own contract, please redeploy it with the following `view` functions:
```
[the remark this answers is not in the document]
Comment (briandoyle81CB) [reply apps/base-docs/base-camp/docs/reading-and-displaying-data/useContractRead.md:31]: We're assuming the learner knows how at this point, and earlier up we linked to the original exercise.  We've also got a copy of a contract they can use down below!

**Review by taycaldwell — APPROVED** [review]
